accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,213 @@
+- Feature Name: Hub Retail Extension
+- Start Date: 2020-11-30
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC describes an extension of retail features for Hub environment.
+
+# Motivation
+[motivation]: #motivation
+
+Reasons for use of Hub with Retail environment
+----------------------------------------------
+
+- scalability
+  - support large environments with the number of terminal exceeding limits of one SUMA server
+
+- disconnected operation
+  - Peripheral server must stay fully functional when the Hub connection is down
+
+
+Possible configurations
+-----------------------
+
+1. Full
+```
+    Terminal -> Branch Server -> Peripheral Server -> Hub
+```
+
+2. Combined SUMA and Branch server 
+```
+    Terminal -> Branch + Peripheral Server Combined -> Hub
+```
+
+3. Standard
+```
+    Terminal -> Branch Server -> SUMA Server
+```
+
+4. Single server
+```
+    Terminal -> Branch + SUMA Server Combined
+```
+
+
+Global Retail data
+------------------
+
+The following items are typically shared in whole organization, so they should be visible and 
+manageable via Hub.
+
+  - Images - distributed as large tarballs via http
+  - Image metadata - pillars, list of packages
+  - HWTYPE configurations - group formula
+  
+Local data
+----------
+
+The following items do not need central management and can stay on Peripheral Server.
+
+  - Branch groups -  groups containing Branch Server and connected terminals
+  - Branch formulas - describe configuration of Branch services - network, pxe, dhcp, dns
+  - Branch PXE data - entries describing how to boot each terminal
+  
+# Detailed design
+[design]: #detailed-design
+
+Images + Image metadata
+-----------------------
+
+Current situation: an image is built on Buildhost, then pushed to SUMA Server htdocs directory. Then SUMA creates a db entry (including package list) and a pillar file.
+The pillar conains URL of image tarball, checksums and additional deployment details. It is used for image synchronization and deployment. These data are extracted 
+on Buildhost at the end of build phase. Some of the data can't be extracted later form the tarball.
+
+The SUMA API related to images is rather docker-centered. For Kiwi images it allows to list images and packages and delete an image.
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
```

Comment:
Branch server downloads the image tarball and then serves it to terminals.
This is about Branch server downloading the tarball.
Now it downloads it from SUMA http server, but any http server can work.
For large environment it can be dedicated http server, or maybe public cloud.

Reply:
I am not sure if the "ISS only" mechanism fits the use case with Lab server
https://github.com/uyuni-project/uyuni-rfc/pull/45/files#diff-1d42c41712a386afc3dd36a0fd7ce312404a01ccd1a535b4e994cdfeb1a3927cR134

Reply:
@nadvornik I would expect that case to be something we can cover as well. Just as we move software channels by copying RPMs and their associated database metadata around without changing them, we should be able to move images and their associated metadata as well.